src/Cms/Core.php

```diff
@@ -310,13 +310,22 @@ public function load(): Loader
 	 */
 	public function roots(): array
 	{
+		$indexClosure = function (array $roots) {
+			// prevent PHPUnit tests from accessing files outside the repo
+			if (defined('KIRBY_TESTING') === true && KIRBY_TESTING === true) {
+				return '/dev/null';
+			}
+
+			return dirname(__DIR__, 3); // @codeCoverageIgnore
+		};
```

Comment:
You see any chance how we could keep this test-specific code out of our core classes?

Reply:
Alternatively if this has to be here, we could move the check to a new `App::isRunningUnitTests(): bool` and then do

```php
'index'       => fn () => match ($this->isRunningUnitTests()) {
    true  => '/dev/null', // @codeCoverageIgnore
    false => dirname(__DIR__, 3); // @codeCoverageIgnore
}
```

Reply:
I pushed another suggestion that at least keeps the "testing mode" logic out of the core.